components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pap/store/PAPPolicyStoreReader.java

```diff
@@ -188,125 +126,42 @@ public boolean isExistPolicy(String policyId) {
      */
     public PolicyDTO readMetaDataPolicyDTO(String policyId) throws EntitlementException {
-        Resource resource = null;
-        PolicyDTO dto = null;
 
-        resource = store.getPolicy(policyId, PDPConstants.ENTITLEMENT_POLICY_PAP);
-        if (resource == null) {
+        PolicyDTO dto = store.getPAPPolicy(policyId);
+        if (dto == null) {
             return null;
         }
-        dto = new PolicyDTO();
-        dto.setPolicyId(policyId);
-        dto.setActive(Boolean.parseBoolean(resource.getProperty(PDPConstants.ACTIVE_POLICY)));
-        String policyOrder = resource.getProperty(PDPConstants.POLICY_ORDER);
-        if (policyOrder != null) {
-            dto.setPolicyOrder(Integer.parseInt(policyOrder));
-        } else {
-            dto.setPolicyOrder(0);
-        }
-
-        String version = resource.getProperty(PDPConstants.POLICY_VERSION);
-        if (version != null) {
-            dto.setVersion(version);
-        }
-        String lastModifiedTime = resource.getProperty(PDPConstants.LAST_MODIFIED_TIME);
-        if (lastModifiedTime != null) {
-            dto.setLastModifiedTime(lastModifiedTime);
-        }
-        String lastModifiedUser = resource.getProperty(PDPConstants.LAST_MODIFIED_USER);
-        if (lastModifiedUser != null) {
-            dto.setLastModifiedUser(lastModifiedUser);
-        }
-        dto.setPolicyType(resource.getProperty(PDPConstants.POLICY_TYPE));
-
-        String policyReferences = resource.getProperty(PDPConstants.POLICY_REFERENCE);
-        if (policyReferences != null && policyReferences.trim().length() > 0) {
-            dto.setPolicyIdReferences(policyReferences.split(PDPConstants.ATTRIBUTE_SEPARATOR));
-        }
-
-        String policySetReferences = resource.getProperty(PDPConstants.POLICY_SET_REFERENCE);
-        if (policySetReferences != null && policySetReferences.trim().length() > 0) {
-            dto.setPolicySetIdReferences(policySetReferences.split(PDPConstants.ATTRIBUTE_SEPARATOR));
-        }
-
-        dto.setPolicyEditor(resource.getProperty(PDPConstants.POLICY_EDITOR_TYPE));
-        String basicPolicyEditorMetaDataAmount = resource.getProperty(PDPConstants.
-                BASIC_POLICY_EDITOR_META_DATA_AMOUNT);
-        if (basicPolicyEditorMetaDataAmount != null) {
-            int amount = Integer.parseInt(basicPolicyEditorMetaDataAmount);
-            String[] basicPolicyEditorMetaData = new String[amount];
-            for (int i = 0; i < amount; i++) {
-                basicPolicyEditorMetaData[i] = resource.
-                        getProperty(PDPConstants.BASIC_POLICY_EDITOR_META_DATA + i);
-            }
-            dto.setPolicyEditorData(basicPolicyEditorMetaData);
-        }
-        PolicyAttributeBuilder policyAttributeBuilder = new PolicyAttributeBuilder();
-        dto.setAttributeDTOs(policyAttributeBuilder.
-                getPolicyMetaDataFromRegistryProperties(resource.getProperties()));
+        dto.setPolicy(null);
```

Comment:
Have to validate, whether this approach (pulling the policy from the database and setting the policy as null in the runtime) will actually serve the purpose of building "Light weight" policy DTO.

Reply:
Refer https://github.com/wso2/carbon-identity-framework/pull/5686#discussion_r1620390382